lib/ansible/plugins/inventory/onesixtyone.py

```diff
@@ -0,0 +1,244 @@
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: onesixtyone
+    plugin_type: inventory
+    version_added: "2.8"
+    short_description: Uses onesixtyone to find hosts to target
+    description:
+        - Uses any YAML file which ends with onesixtyone.yml or onesixtyone.yaml as a configuration file.
+    extends_documentation_fragment:
+      - constructed
+      - inventory_cache
+    requirements:
+      - onesixtyone CLI installed
+      - python-gevent installed
+      - netaddr python library installed
+    options:
+        addresses:
+            description: List of CIDRs and IPs to scan.
+            type: list
+            required: false
+            default:
+              - 192.168.0.0/24
+        communities:
+            description: List of communities to query per device.
+            type: list
+            required: false
+            default:
+              - "public"
+        wait:
+          description: wait n milliseconds (1/1000 of a second) between sending packets.
+          type: int
+          default: 10
+          required: false
+        platform:
+          description: Dict containing additional mappings between Platform (key) and regex to match in sysDescr (value).
+          type: dict
+          default: {}
+          required: false
+    notes:
+      - Only discover devices running SNMPv2.
+      - 'TODO: Add caching support.'
+'''
+EXAMPLES = '''
+    # onesixtyone.yaml or any other YAML file-extension
+    plugin: onesixtyone
+    strict: False
+    addresses:
+      - 192.168.0.0/24
+'''
+
+import os
+import re
+import tempfile
+import subprocess
+
+from ansible import constants as C
+from ansible.errors import AnsibleParserError
+from ansible.module_utils._text import to_native
+from ansible.module_utils.common.process import get_bin_path
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+
+try:
+    from netaddr import IPSet
+    HAS_NETADDR = True
+except Exception:
+    HAS_NETADDR = False
+
+try:
+    import gevent
+    from gevent import socket
+    HAS_GEVENT = True
+except Exception:
+    HAS_GEVENT = False
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+    NAME = 'onesixtyone'
+
+    def __init__(self):
+        super(InventoryModule, self).__init__()
+
+    def verify_file(self, path):
+        """ Verify the existence of plugin's configuration file """
+        if not super(InventoryModule, self).verify_file(path):
+            return False
+
+        for suffix in ('yaml', 'yml'):
+            maybe = '{fn}.{suffix}'.format(fn=self.NAME, suffix=suffix)
+            if path.endswith(maybe):
+                return True
+
+    def parse(self, inventory, loader, path, cache=False):
+        super(InventoryModule, self).parse(inventory, loader, path, cache=cache)
+        self._onesixtyone = get_bin_path(self.NAME, True)
```

Comment:
this produces an exception when binary cannot be found, you should capture and return parser error

Reply:
Moved it under the `try` in `parser` method.